# C5.3 Problems
<hr style="height:5px;border-width:0;color:gray;background-color:gray">

## Problem C5.1 — Magnetic Force on a Straight Wire in a Uniform Field

A straight copper wire segment of length

$$
L = 0.60\ \text{m}
$$

lies along the **positive $y$-axis** and carries a steady current

$$
I = 8.0\ \text{A}
$$

flowing upward.

The wire is placed in a uniform magnetic field of magnitude

$$
B = 0.35\ \text{T}.
$$

The magnetic field lies in the $xy$-plane and points **$30^\circ$ clockwise from the positive $x$-axis** (i.e., toward the negative $y$ direction).

Answer the following:

1. Determine the angle $\theta$ between the current direction and the magnetic field.
2. Compute the magnitude of the magnetic force on the wire.
3. Determine the direction of the magnetic force (state clearly whether it points into or out of the page).
4. Suppose the current is reversed. What happens to the force direction?

Be careful with geometry and provide a short physical interpretation of your results.

<details>
<summary style="background-color:#006633; color:white; padding:8px; border-radius:4px; cursor:pointer;">
Problem solution
</summary>

<div style="background-color:#e8f5e9; padding:10px; border-radius:4px; margin-top:6px;">

**Relevant equation**

Magnetic force on a straight wire:

$$
\vec{F} = I\,\vec{L}\times\vec{B}
$$

Magnitude form:

$$
F = I L B \sin\theta
$$

---

**Given**

- $L = 0.60\ \text{m}$
- $I = 8.0\ \text{A}$
- $B = 0.35\ \text{T}$
- Current along $+y$
- $\vec{B}$ is $30^\circ$ clockwise from $+x$

---

**1) Determine the angle $\theta$**

The current direction is along $+y$.

A magnetic field that is $30^\circ$ clockwise from $+x$ has:

- a positive $x$ component  
- a negative $y$ component  

Thus, the angle between $+y$ and $\vec{B}$ is

$$
\theta = 90^\circ + 30^\circ = 120^\circ.
$$

---

**2) Magnitude of the magnetic force**

$$
F = (8.0)(0.60)(0.35)\sin(120^\circ).
$$

Since

$$
\sin(120^\circ) = \sin(60^\circ) = 0.866,
$$

$$
F = (8.0)(0.60)(0.35)(0.866).
$$

$$
F = 1.46\ \text{N}.
$$

$$
\boxed{F = 1.46\ \text{N}}
$$

---

**3) Direction of the force**

Using

$$
\vec{F} = I\,\vec{L}\times\vec{B},
$$

- $\vec{L}$ points along $+y$.
- $\vec{B}$ has a positive $x$ component and a negative $y$ component.

The perpendicular component of $\vec{B}$ is primarily along $+x$.

Using the right-hand rule:

$$
\hat{y} \times \hat{x} = -\hat{z}.
$$

Therefore, the force points **into the page ($-z$ direction)**.

---

**4) If the current reverses**

If the current flows downward ($-y$ direction),

the cross product reverses sign.

The force would then point **out of the page ($+z$ direction)**.

---

**Physical interpretation**

- Only the component of $\vec{B}$ perpendicular to the current contributes to the force.
- Even when $\theta > 90^\circ$, the sine function ensures only perpendicular geometry determines magnitude.
- Reversing the current reverses the force direction — a key mechanism in electromagnetic devices.

</div>
</details>

---


In [1]:
# DIY Cell

In [2]:
%reset -f

import sympy as sp
from IPython.display import display, Math

# -----------------------------
# Given values
# -----------------------------
L = 0.60      # m
I = 8.0       # A
B = 0.35      # T
theta_deg = 120.0  # degrees (angle between +y current direction and B at 30° clockwise from +x)

# -----------------------------
# Physics:
# Magnetic force on a straight wire in a uniform B-field:
#
# Vector form:
#   F⃗ = I (L⃗ × B⃗)
#
# Magnitude:
#   F = I L B sin(theta)
#
# Direction:
#   Determined by the right-hand rule for L⃗ × B⃗.
#   With L⃗ along +y and B⃗ having a +x component, ŷ × x̂ = -ẑ (into the page).
#   Reversing current flips L⃗ → -L⃗ and flips the force direction.
# -----------------------------
theta = sp.rad(theta_deg)
F = I * L * B * sp.sin(theta)

# Numeric value
F_num = float(F.evalf())

# -----------------------------
# Nicely formatted output
# -----------------------------
print("Magnetic force on a straight current-carrying wire\n")
print("Given:")
print(f"  Wire length:       L = {L:.3e} m")
print(f"  Current:           I = {I:.3e} A")
print(f"  Magnetic field:    B = {B:.3e} T")
print(f"  Angle:           theta = {theta_deg:.3g} degrees\n")

print("Results (3 significant figures):")
print(f"  Force magnitude:   F = {F_num:.3e} N\n")

print("Direction (Right-Hand Rule):")
print("• Let the wire lie along +y (current upward).")
print("• The magnetic field lies in the xy-plane, 30° clockwise from +x, so it has a +x component.")
print("• Using the cross product direction: ŷ × x̂ = -ẑ.")
print("  → The force points into the page (−z direction).")
print("• If the current is reversed (along −y), the force reverses direction:")
print("  → The force points out of the page (+z direction).\n")

print("Physical interpretation:")
print("• Only the component of B perpendicular to the wire contributes through sin(theta).")
print("• The force is perpendicular to both the current direction and the magnetic field direction.")
print("• Reversing the current reverses the force direction (a key mechanism in motors and actuators).")


Magnetic force on a straight current-carrying wire

Given:
  Wire length:       L = 6.000e-01 m
  Current:           I = 8.000e+00 A
  Magnetic field:    B = 3.500e-01 T
  Angle:           theta = 120 degrees

Results (3 significant figures):
  Force magnitude:   F = 1.455e+00 N

Direction (Right-Hand Rule):
• Let the wire lie along +y (current upward).
• The magnetic field lies in the xy-plane, 30° clockwise from +x, so it has a +x component.
• Using the cross product direction: ŷ × x̂ = -ẑ.
  → The force points into the page (−z direction).
• If the current is reversed (along −y), the force reverses direction:
  → The force points out of the page (+z direction).

Physical interpretation:
• Only the component of B perpendicular to the wire contributes through sin(theta).
• The force is perpendicular to both the current direction and the magnetic field direction.
• Reversing the current reverses the force direction (a key mechanism in motors and actuators).


<hr style="height:5px;border-width:0;color:gray;background-color:gray">

## Problem C5.2 — Magnetic Force on a Straight Wire

A straight wire segment of length

$$
L = 0.25\ \text{m}
$$

is oriented along the direction

$$
\hat{\ell}=\frac{1}{\sqrt{14}}\left(1\,\hat{x}+2\,\hat{y}+3\,\hat{z}\right).
$$

A steady current of

$$
I = 7.0\ \text{A}
$$

flows through the wire in the direction of $\hat{\ell}$.

The wire is placed in a uniform magnetic field given in vector form by

$$
\vec{B} = \left(0.20\,\hat{x}-0.10\,\hat{y}+0.30\,\hat{z}\right)\ \text{T}.
$$

Answer the following:

1. Write the wire length vector $\vec{L}$ in component form.
2. Compute the magnetic force vector using

   $$
   \vec{F} = I\,\vec{L}\times\vec{B}.
   $$

3. Report $\vec{F}$ in unit-vector form ($\hat{x},\hat{y},\hat{z}$) and compute its magnitude $F$.
4. State one clear physical interpretation of the direction of $\vec{F}$ relative to $\vec{L}$ and $\vec{B}$.

<details>
<summary style="background-color:#006633; color:white; padding:8px; border-radius:4px; cursor:pointer;">
Problem solution
</summary>

<div style="background-color:#e8f5e9; padding:10px; border-radius:4px; margin-top:6px;">

**Relevant equation**

$$
\vec{F} = I\,\vec{L}\times\vec{B}.
$$

---

**Given**

- $L = 0.25\ \text{m}$
- $I = 7.0\ \text{A}$
- $\hat{\ell}=\dfrac{1}{\sqrt{14}}\left(1\,\hat{x}+2\,\hat{y}+3\,\hat{z}\right)$
- $\vec{B} = \left(0.20\,\hat{x}-0.10\,\hat{y}+0.30\,\hat{z}\right)\ \text{T}$

---

**1) Wire length vector $\vec{L}$**

$$
\vec{L} = L\,\hat{\ell}
= 0.25\cdot \frac{1}{\sqrt{14}}\left(1\,\hat{x}+2\,\hat{y}+3\,\hat{z}\right).
$$

So

$$
\vec{L}=\left(\frac{0.25}{\sqrt{14}}\right)\hat{x}
+\left(\frac{0.50}{\sqrt{14}}\right)\hat{y}
+\left(\frac{0.75}{\sqrt{14}}\right)\hat{z}.
$$

Let

$$
a=\frac{0.25}{\sqrt{14}},\quad b=\frac{0.50}{\sqrt{14}},\quad c=\frac{0.75}{\sqrt{14}}.
$$

So $\vec{L}=a\hat{x}+b\hat{y}+c\hat{z}$ and $\vec{B}=0.20\hat{x}-0.10\hat{y}+0.30\hat{z}$.

---

**2) Compute $\vec{L}\times\vec{B}$ by expanding**

Distribute the cross product:

$$
\vec{L}\times\vec{B}
=
(a\hat{x}+b\hat{y}+c\hat{z})\times(0.20\hat{x}-0.10\hat{y}+0.30\hat{z}).
$$

Expand term-by-term:

$$
\vec{L}\times\vec{B}
=
a\hat{x}\times(0.20\hat{x})
+a\hat{x}\times(-0.10\hat{y})
+a\hat{x}\times(0.30\hat{z})
$$

$$
\quad
+b\hat{y}\times(0.20\hat{x})
+b\hat{y}\times(-0.10\hat{y})
+b\hat{y}\times(0.30\hat{z})
$$

$$
\quad
+c\hat{z}\times(0.20\hat{x})
+c\hat{z}\times(-0.10\hat{y})
+c\hat{z}\times(0.30\hat{z}).
$$

Now use the unit-vector rules:

- $\hat{x}\times\hat{x}=0$, $\hat{y}\times\hat{y}=0$, $\hat{z}\times\hat{z}=0$
- $\hat{x}\times\hat{y}=\hat{z}$, $\hat{y}\times\hat{x}=-\hat{z}$
- $\hat{x}\times\hat{z}=-\hat{y}$, $\hat{z}\times\hat{x}=\hat{y}$
- $\hat{y}\times\hat{z}=\hat{x}$, $\hat{z}\times\hat{y}=-\hat{x}$

Compute the nonzero terms:

- $a\hat{x}\times(-0.10\hat{y}) = -0.10a(\hat{x}\times\hat{y}) = -0.10a\,\hat{z}$
- $a\hat{x}\times(0.30\hat{z}) = 0.30a(\hat{x}\times\hat{z}) = 0.30a(-\hat{y}) = -0.30a\,\hat{y}$
- $b\hat{y}\times(0.20\hat{x}) = 0.20b(\hat{y}\times\hat{x}) = 0.20b(-\hat{z}) = -0.20b\,\hat{z}$
- $b\hat{y}\times(0.30\hat{z}) = 0.30b(\hat{y}\times\hat{z}) = 0.30b\,\hat{x}$
- $c\hat{z}\times(0.20\hat{x}) = 0.20c(\hat{z}\times\hat{x}) = 0.20c\,\hat{y}$
- $c\hat{z}\times(-0.10\hat{y}) = -0.10c(\hat{z}\times\hat{y}) = -0.10c(-\hat{x}) = +0.10c\,\hat{x}$

Group like unit vectors:

**$\hat{x}$ component**

$$
(\vec{L}\times\vec{B})_x = 0.30b + 0.10c
$$

**$\hat{y}$ component**

$$
(\vec{L}\times\vec{B})_y = -0.30a + 0.20c
$$

**$\hat{z}$ component**

$$
(\vec{L}\times\vec{B})_z = -0.10a - 0.20b
$$

Now substitute $a,b,c$:

$$
(\vec{L}\times\vec{B})_x
= 0.30\left(\frac{0.50}{\sqrt{14}}\right)
+0.10\left(\frac{0.75}{\sqrt{14}}\right)
=\frac{0.150+0.075}{\sqrt{14}}
=\frac{0.225}{\sqrt{14}}.
$$

$$
(\vec{L}\times\vec{B})_y
= -0.30\left(\frac{0.25}{\sqrt{14}}\right)
+0.20\left(\frac{0.75}{\sqrt{14}}\right)
=\frac{-0.075+0.150}{\sqrt{14}}
=\frac{0.075}{\sqrt{14}}.
$$

$$
(\vec{L}\times\vec{B})_z
= -0.10\left(\frac{0.25}{\sqrt{14}}\right)
-0.20\left(\frac{0.50}{\sqrt{14}}\right)
=\frac{-0.025-0.100}{\sqrt{14}}
=-\frac{0.125}{\sqrt{14}}.
$$

So

$$
\vec{L}\times\vec{B}
=
\frac{1}{\sqrt{14}}
\left(0.225\,\hat{x}+0.075\,\hat{y}-0.125\,\hat{z}\right).
$$

---

**3) Force vector and magnitude**

Multiply by $I=7.0$:

$$
\vec{F}
=
\frac{7.0}{\sqrt{14}}
\left(0.225\,\hat{x}+0.075\,\hat{y}-0.125\,\hat{z}\right).
$$

Numerically, since $\sqrt{14}\approx 3.74$,

$$
\vec{F}\approx
(0.421\,\hat{x}+0.140\,\hat{y}-0.234\,\hat{z})\ \text{N}.
$$

Magnitude:

$$
F=\sqrt{F_x^2+F_y^2+F_z^2}
\approx \sqrt{(0.421)^2+(0.140)^2+(-0.234)^2}.
$$

$$
F \approx 0.50\ \text{N}.
$$

---

**4) Physical interpretation**

- The force is perpendicular to both $\vec{L}$ (current direction) and $\vec{B}$ because it comes from a cross product.
- If you reverse the current direction ($\vec{L}\to -\vec{L}$), then $\vec{F}$ reverses direction.
- The wire is pushed sideways relative to both the current direction and the magnetic field direction.

</div>
</details>

---


In [3]:
# DIY Cell

In [4]:
%reset -f

import sympy as sp
from IPython.display import display

# -----------------------------
# Given values
# -----------------------------
L = 0.25     # m
I = 7.0      # A

# Magnetic field components (T)
Bx = 0.20
By = -0.10
Bz = 0.30

# -----------------------------
# Define symbolic unit vectors
# -----------------------------
xhat = sp.Matrix([1,0,0])
yhat = sp.Matrix([0,1,0])
zhat = sp.Matrix([0,0,1])

# -----------------------------
# Wire direction (unit vector)
# ℓ̂ = (1/√14)(1 x̂ + 2 ŷ + 3 ẑ)
# -----------------------------
ell_hat = (1/sp.sqrt(14)) * sp.Matrix([1,2,3])

# Wire length vector
L_vec = L * ell_hat

# Magnetic field vector
B_vec = sp.Matrix([Bx, By, Bz])

# -----------------------------
# Magnetic force:
# F⃗ = I (L⃗ × B⃗)
# -----------------------------
F_vec = I * L_vec.cross(B_vec)

# Magnitude
F_mag = sp.sqrt(F_vec.dot(F_vec))

# Numeric evaluation
F_vec_num = F_vec.evalf()
F_mag_num = F_mag.evalf()

# -----------------------------
# Output
# -----------------------------
print("Magnetic force on a straight wire (3D cross product)\n")

print("Given:")
print(f"  Wire length L = {L:.3e} m")
print(f"  Current I = {I:.3e} A")
print(f"  Magnetic field B = ({Bx:.2f}, {By:.2f}, {Bz:.2f}) T\n")

print("Wire length vector L⃗:")
display(L_vec)

print("\nMagnetic force vector F⃗ = I (L⃗ × B⃗):")
display(F_vec_num)

print(f"\nForce magnitude |F⃗| = {F_mag_num:.3e} N\n")

print("Physical interpretation:")
print("• The force is perpendicular to both L⃗ (current direction) and B⃗.")
print("• Reversing the current would reverse the direction of F⃗.")
print("• The magnitude depends on the component of B⃗ perpendicular to the wire.")


Magnetic force on a straight wire (3D cross product)

Given:
  Wire length L = 2.500e-01 m
  Current I = 7.000e+00 A
  Magnetic field B = (0.20, -0.10, 0.30) T

Wire length vector L⃗:


Matrix([
[0.0178571428571429*sqrt(14)],
[0.0357142857142857*sqrt(14)],
[0.0535714285714286*sqrt(14)]])


Magnetic force vector F⃗ = I (L⃗ × B⃗):


Matrix([
[ 0.420936456012068],
[ 0.140312152004023],
[-0.233853586673371]])


Force magnitude |F⃗| = 5.016e-1 N

Physical interpretation:
• The force is perpendicular to both L⃗ (current direction) and B⃗.
• Reversing the current would reverse the direction of F⃗.
• The magnitude depends on the component of B⃗ perpendicular to the wire.


<hr style="height:5px;border-width:0;color:gray;background-color:gray">

## Problem C5.3 — Net Magnetic Force on a Curved Conductor (Using $d\vec{F}$ and Integration)

A wire is bent into a **quarter-circle arc** of radius

$$
R = 0.30\ \text{m}
$$

lying in the $xy$-plane, centered at the origin. The arc runs from the point $(R,0,0)$ to $(0,R,0)$.

A steady current

$$
I = 4.0\ \text{A}
$$

flows **counterclockwise** along the arc (from $(R,0,0)$ toward $(0,R,0)$).

The wire is placed in a uniform magnetic field

$$
\vec{B} = \left(0.50\,\hat{x} + 0.20\,\hat{y} + 0.00\,\hat{z}\right)\ \text{T}.
$$

Answer the following:

1. Parameterize the arc using an angle $\theta$ and write $\vec{r}(\theta)$ for the wire.
2. Compute the differential length vector $d\vec{\ell}$.
3. Use

   $$
   d\vec{F} = I\, d\vec{\ell} \times \vec{B}
   $$

   to write an expression for $d\vec{F}$ in unit-vector form.
4. Integrate to find the net force $\vec{F}_{\text{net}}$ on the quarter-circle arc.
5. Compute the magnitude $F_{\text{net}}$ and state the direction (sign of the $z$-component).

<details>
<summary style="background-color:#006633; color:white; padding:8px; border-radius:4px; cursor:pointer;">
Problem solution
</summary>

<div style="background-color:#e8f5e9; padding:10px; border-radius:4px; margin-top:6px;">

**Relevant equation**

$$
\vec{F}_{\text{net}} = I\int d\vec{\ell}\times\vec{B}.
$$

---

**Given**

- $R=0.30\ \text{m}$
- $I=4.0\ \text{A}$
- $\vec{B}=(0.50\,\hat{x}+0.20\,\hat{y})\ \text{T}$
- Quarter-circle arc in the $xy$-plane, CCW from $\theta=0$ to $\theta=\frac{\pi}{2}$

---

**1) Parameterize the arc**

A quarter circle of radius $R$ in the $xy$-plane:

$$
\vec{r}(\theta)=R\cos\theta\,\hat{x}+R\sin\theta\,\hat{y},
\qquad 0\le \theta \le \frac{\pi}{2}.
$$

---

**2) Differential length vector**

$$
d\vec{\ell} = \frac{d\vec{r}}{d\theta}\,d\theta
= \left(-R\sin\theta\,\hat{x}+R\cos\theta\,\hat{y}\right)d\theta.
$$

---

**3) Compute $d\vec{F}$**

Let

$$
\vec{B}=B_x\hat{x}+B_y\hat{y},\quad B_x=0.50,\ B_y=0.20.
$$

Then

$$
d\vec{F}=I\,d\vec{\ell}\times\vec{B}
=
I\left(-R\sin\theta\,\hat{x}+R\cos\theta\,\hat{y}\right)d\theta
\times\left(B_x\hat{x}+B_y\hat{y}\right).
$$

Expand using unit-vector rules:

- $\hat{x}\times\hat{x}=0$, $\hat{y}\times\hat{y}=0$
- $\hat{x}\times\hat{y}=\hat{z}$
- $\hat{y}\times\hat{x}=-\hat{z}$

So

$$
d\vec{F}
= I R d\theta\left[(-\sin\theta)B_y(\hat{x}\times\hat{y}) + (\cos\theta)B_x(\hat{y}\times\hat{x})\right].
$$

$$
d\vec{F}
= I R d\theta\left[(-\sin\theta)B_y\,\hat{z} + (\cos\theta)B_x(-\hat{z})\right].
$$

$$
d\vec{F}
= -IR\left(B_y\sin\theta + B_x\cos\theta\right)\hat{z}\,d\theta.
$$

---

**4) Integrate to find $\vec{F}_{\text{net}}$**

$$
\vec{F}_{\text{net}}
=
\int_0^{\pi/2} d\vec{F}
=
-IR\hat{z}\int_0^{\pi/2}\left(B_y\sin\theta + B_x\cos\theta\right)d\theta.
$$

Evaluate:

$$
\int_0^{\pi/2} B_y\sin\theta\,d\theta
=
B_y\left[-\cos\theta\right]_0^{\pi/2}
=
B_y(1).
$$

$$
\int_0^{\pi/2} B_x\cos\theta\,d\theta
=
B_x\left[\sin\theta\right]_0^{\pi/2}
=
B_x(1).
$$

So

$$
\vec{F}_{\text{net}} = -IR(B_x+B_y)\,\hat{z}.
$$

Substitute numbers:

$$
\vec{F}_{\text{net}}
= -(4.0)(0.30)\left(0.50+0.20\right)\hat{z}
= -0.84\,\hat{z}\ \text{N}.
$$

$$
\boxed{\vec{F}_{\text{net}} = -0.84\,\hat{z}\ \text{N}}
$$

---

**5) Magnitude and direction**

Magnitude:

$$
F_{\text{net}} = 0.84\ \text{N}.
$$

Direction:

- Negative $z$ means the force points **into the page**.

---

**Physical interpretation**

- Every small piece of wire feels a force perpendicular to both $d\vec{\ell}$ and $\vec{B}$.
- Here the geometry causes all contributions to add in the $-z$ direction, producing a net force.
- This is exactly why curved conductors can have a nonzero net force even in a uniform field when the conductor is not a closed loop.

</div>
</details>

---


In [ ]:
# DIY Cell

In [5]:
%reset -f

import sympy as sp
from IPython.display import display

# -----------------------------
# Given values
# -----------------------------
R = 0.30      # m
I = 4.0       # A

# Magnetic field vector components (T)
Bx = 0.50
By = 0.20
Bz = 0.00

# -----------------------------
# Symbols and vectors
# -----------------------------
theta = sp.symbols('theta', real=True)

# r(θ) for quarter-circle arc in xy-plane, 0 ≤ θ ≤ π/2
r = sp.Matrix([R*sp.cos(theta), R*sp.sin(theta), 0])

# dℓ = dr/dθ dθ
dr_dtheta = sp.diff(r, theta)

# Magnetic field vector
B = sp.Matrix([Bx, By, Bz])

# Differential force per dθ: dF/dθ = I (dr/dθ × B)
dF_dtheta = I * dr_dtheta.cross(B)

# Net force: F = ∫ (dF/dθ) dθ from 0 to π/2
F_net = sp.integrate(dF_dtheta, (theta, 0, sp.pi/2))

# Magnitude
F_mag = sp.sqrt(F_net.dot(F_net))

# Numeric evaluation
F_net_num = sp.N(F_net)
F_mag_num = sp.N(F_mag)

# -----------------------------
# Output
# -----------------------------
print("Net magnetic force on a quarter-circle arc (C5.3)\n")

print("Given:")
print(f"  Radius:            R = {R:.3e} m")
print(f"  Current:           I = {I:.3e} A")
print(f"  Magnetic field:    B = ({Bx:.2f}, {By:.2f}, {Bz:.2f}) T\n")

print("Differential force (per dθ): dF/dθ = I (dr/dθ × B)")
display(dF_dtheta)

print("\nNet force vector F_net = ∫ dF from 0 to π/2:")
display(F_net)

print("\nNumeric net force vector:")
display(F_net_num)

print(f"\nForce magnitude |F_net| = {F_mag_num:.3e} N\n")

print("Direction check:")
print("• The result is along the z-axis.")
print("• A negative z-component means the force points into the page (−z).")


Net magnetic force on a quarter-circle arc (C5.3)

Given:
  Radius:            R = 3.000e-01 m
  Current:           I = 4.000e+00 A
  Magnetic field:    B = (0.50, 0.20, 0.00) T

Differential force (per dθ): dF/dθ = I (dr/dθ × B)


Matrix([
[                                0],
[                                0],
[-0.24*sin(theta) - 0.6*cos(theta)]])


Net force vector F_net = ∫ dF from 0 to π/2:


Matrix([
[    0],
[    0],
[-0.84]])


Numeric net force vector:


Matrix([
[    0],
[    0],
[-0.84]])


Force magnitude |F_net| = 8.400e-1 N

Direction check:
• The result is along the z-axis.
• A negative z-component means the force points into the page (−z).


<hr style="height:5px;border-width:0;color:gray;background-color:gray">

## Problem C5.4 — Force and Torque on a Current-Carrying Rod

A straight conducting rod of length

$$
L = 0.40\ \text{m}
$$

lies along the $x$-axis from $x=0$ to $x=L$ in the horizontal $xy$-plane. A steady current

$$
I = 3.5\ \text{A}
$$

flows in the $+x$ direction.

The rod is placed in a magnetic field that points upward and varies with position:

$$
\vec{B}(x) = B(x)\,\hat{z},\qquad B(x)=\big(0.20 + 0.80x\big)\ \text{T},
$$

where $x$ is in meters.

The rod is free to rotate about a pivot at the origin ($x=0$) about the $z$-axis.

Use the differential force law

$$
d\vec{F} = I\,d\vec{\ell}\times\vec{B}(x),
$$

and the differential torque about the pivot

$$
d\vec{\tau} = \vec{r}\times d\vec{F}.
$$

Answer the following:

1. Write $d\vec{\ell}$ and $\vec{r}$ for this geometry.
2. Determine $d\vec{F}$ as a function of $x$ (include both magnitude and direction).
3. Integrate to find the **net force** $\vec{F}_{\text{net}}$ on the rod.
4. Integrate to find the **net torque** about the pivot, $\tau_z$.
5. Give a short physical interpretation: why does a non-uniform field lead to a nonzero torque here?

---

<details>
<summary style="background-color:#006633; color:white; padding:8px; border-radius:4px; cursor:pointer;">
Problem solution
</summary>

<div style="background-color:#e8f5e9; padding:10px; border-radius:4px; margin-top:6px;">

**Relevant equations**

$$
d\vec{F} = I\,d\vec{\ell}\times\vec{B}(x),
\qquad
\vec{F}_{\text{net}}=\int d\vec{F}.
$$

$$
d\vec{\tau}=\vec{r}\times d\vec{F},
\qquad
\tau_z = \int d\tau_z.
$$

---

**Given**

- Rod along $+x$ from $0$ to $L=0.40\ \text{m}$
- $I=3.5\ \text{A}$ in $+x$
- $\vec{B}(x)=B(x)\hat{z}$ with $B(x)=0.20+0.80x$ (T)

---

**1) Geometry: $d\vec{\ell}$ and $\vec{r}$**

Along the rod,

$$
d\vec{\ell}=dx\,\hat{x}.
$$

A point at location $x$ has position vector (from the pivot at the origin):

$$
\vec{r}=x\,\hat{x}.
$$

---

**2) Differential force**

$$
d\vec{F}=I\,(dx\,\hat{x})\times\big(B(x)\hat{z}\big)
=I\,B(x)\,dx\,(\hat{x}\times\hat{z}).
$$

Using $\hat{x}\times\hat{z}=-\hat{y}$,

$$
d\vec{F}=-I\,B(x)\,dx\,\hat{y}.
$$

So the force on each element points in the $-\hat{y}$ direction.

---

**3) Net force**

$$
\vec{F}_{\text{net}}=\int_0^L d\vec{F}
=-I\hat{y}\int_0^L B(x)\,dx.
$$

Insert $B(x)=0.20+0.80x$:

$$
\int_0^L (0.20+0.80x)\,dx
=\left[0.20x+0.40x^2\right]_0^L
=0.20L+0.40L^2.
$$

With $L=0.40$:

$$
0.20L+0.40L^2
=0.20(0.40)+0.40(0.40^2)
=0.080+0.064
=0.144.
$$

Thus

$$
\vec{F}_{\text{net}}=-I(0.144)\,\hat{y}
=-(3.5)(0.144)\,\hat{y}.
$$

$$
\boxed{\vec{F}_{\text{net}}=-0.504\,\hat{y}\ \text{N}}
$$

---

**4) Net torque about the pivot**

$$
d\vec{\tau}=\vec{r}\times d\vec{F}
=(x\hat{x})\times\left(-I\,B(x)\,dx\,\hat{y}\right)
=-I\,xB(x)\,dx\,(\hat{x}\times\hat{y}).
$$

Since $\hat{x}\times\hat{y}=\hat{z}$,

$$
d\vec{\tau}=-I\,xB(x)\,dx\,\hat{z}.
$$

So

$$
\tau_z=\int_0^L d\tau_z
=-I\int_0^L xB(x)\,dx.
$$

Insert $B(x)=0.20+0.80x$:

$$
\tau_z=-I\int_0^L x(0.20+0.80x)\,dx
=-I\int_0^L (0.20x+0.80x^2)\,dx.
$$

Integrate:

$$
\int_0^L (0.20x+0.80x^2)\,dx
=\left[0.10x^2+\frac{0.80}{3}x^3\right]_0^L
=0.10L^2+\frac{0.80}{3}L^3.
$$

For $L=0.40$:

$$
0.10L^2=0.10(0.16)=0.016,
\qquad
\frac{0.80}{3}L^3=\frac{0.80}{3}(0.064)\approx 0.0171.
$$

Sum:

$$
0.016+0.0171=0.0331.
$$

Thus

$$
\tau_z=-I(0.0331)=-(3.5)(0.0331)=-0.116\ \text{N·m}.
$$

$$
\boxed{\vec{\tau}=-0.116\,\hat{z}\ \text{N·m}}
$$

---

**5) Physical interpretation**

- Each piece of the rod feels a magnetic force in the same $-\hat{y}$ direction.
- Because the rod is pivoted at $x=0$, forces applied farther from the pivot create larger torque ($d\tau\propto x$).
- The magnetic field is stronger at larger $x$, so the outer portion of the rod contributes disproportionately.
- That unequal distribution of force produces a net torque that tends to rotate the rod.

</div>
</details>

---


In [ ]:
# DIY Cell

In [6]:
%reset -f

import sympy as sp
from IPython.display import display

# -----------------------------
# Given values
# -----------------------------
L = 0.40      # m
I = 3.5       # A

# Magnetic field: B(x) = (0.20 + 0.80 x) T, directed +z
x = sp.symbols('x', real=True)
B = 0.20 + 0.80*x

# -----------------------------
# Geometry
# -----------------------------
# Rod lies along +x from 0 to L, so:
# dℓ = dx x̂
# r  = x x̂  (position from pivot at origin)

# Unit vectors as SymPy matrices
xhat = sp.Matrix([1, 0, 0])
yhat = sp.Matrix([0, 1, 0])
zhat = sp.Matrix([0, 0, 1])

dl = xhat            # dℓ/dx (we'll multiply by dx via integration)
r_vec = x * xhat     # position vector

B_vec = B * zhat     # magnetic field vector

# -----------------------------
# Differential force per dx:
# dF = I (dℓ × B) = I [(dℓ/dx) × B] dx
# -----------------------------
dF_dx = I * dl.cross(B_vec)

# Net force: F = ∫ dF = ∫ (dF/dx) dx
F_net = sp.integrate(dF_dx, (x, 0, L))

# -----------------------------
# Differential torque per dx about pivot:
# dτ = r × dF = r × [(dF/dx) dx]
# So dτ/dx = r × (dF/dx)
# -----------------------------
dTau_dx = r_vec.cross(dF_dx)

Tau_net = sp.integrate(dTau_dx, (x, 0, L))

# Magnitudes (optional)
F_mag = sp.sqrt(F_net.dot(F_net))
Tau_mag = sp.sqrt(Tau_net.dot(Tau_net))

# Numeric evaluation
F_net_num = sp.N(F_net)
Tau_net_num = sp.N(Tau_net)
F_mag_num = sp.N(F_mag)
Tau_mag_num = sp.N(Tau_mag)

# -----------------------------
# Output
# -----------------------------
print("C5.3 — Net force and net torque on a current-carrying rod in a non-uniform magnetic field\n")

print("Given:")
print(f"  Rod length:  L = {L:.3e} m")
print(f"  Current:     I = {I:.3e} A")
print("  B(x) = (0.20 + 0.80 x) T in the +z direction, with x in meters\n")

print("Differential force per dx: dF/dx = I[(dℓ/dx) × B(x)]")
display(dF_dx)

print("\nNet force vector F_net = ∫_0^L dF")
display(F_net)
print("Numeric F_net:")
display(F_net_num)
print(f"|F_net| = {F_mag_num:.3e} N\n")

print("Differential torque per dx: dτ/dx = r × (dF/dx)")
display(dTau_dx)

print("\nNet torque vector Tau_net = ∫_0^L dτ")
display(Tau_net)
print("Numeric Tau_net:")
display(Tau_net_num)
print(f"|Tau_net| = {Tau_mag_num:.3e} N·m\n")

print("Direction check:")
print("• F_net should be along −y (since x̂ × ẑ = −ŷ).")
print("• Tau_net should be along −z (since r is +x and F is −y, so x̂ × (−ŷ) = −ẑ).")


C5.3 — Net force and net torque on a current-carrying rod in a non-uniform magnetic field

Given:
  Rod length:  L = 4.000e-01 m
  Current:     I = 3.500e+00 A
  B(x) = (0.20 + 0.80 x) T in the +z direction, with x in meters

Differential force per dx: dF/dx = I[(dℓ/dx) × B(x)]


Matrix([
[           0],
[-2.8*x - 0.7],
[           0]])


Net force vector F_net = ∫_0^L dF


Matrix([
[     0],
[-0.504],
[     0]])

Numeric F_net:


Matrix([
[     0],
[-0.504],
[     0]])

|F_net| = 5.040e-1 N

Differential torque per dx: dτ/dx = r × (dF/dx)


Matrix([
[               0],
[               0],
[x*(-2.8*x - 0.7)]])


Net torque vector Tau_net = ∫_0^L dτ


Matrix([
[                 0],
[                 0],
[-0.115733333333333]])

Numeric Tau_net:


Matrix([
[                 0],
[                 0],
[-0.115733333333333]])

|Tau_net| = 1.157e-1 N·m

Direction check:
• F_net should be along −y (since x̂ × ẑ = −ŷ).
• Tau_net should be along −z (since r is +x and F is −y, so x̂ × (−ŷ) = −ẑ).


<hr style="height:5px;border-width:0;color:gray;background-color:gray">

## Problem C5.5 — Magnetic Actuator Rod: Torque, Angular Acceleration, and Revolutions 

A simple **magnetic actuator** uses a current-carrying arm (a conducting rod) in a magnetic field to produce rotation.  
This is a stripped-down model for the force mechanism inside many devices (e.g., small motors, galvanometers, and electromagnetic actuators).

A straight conducting rod of length

$$
L = 0.20\ \text{m}
$$

is mounted horizontally along the $x$-axis from $x=0$ to $x=L$ and can rotate about a frictionless pivot at $x=0$ (about the $z$-axis). A steady current

$$
I = 2.5\ \text{A}
$$

flows in the $+x$ direction.

The rod sits in a vertical magnetic field that is approximately uniform over the rod:

$$
\vec{B} = B\,\hat{z},\qquad B = 0.60\ \text{T}.
$$

The rotating assembly (rod + lightweight mount) has a moment of inertia about the pivot

$$
\mathcal{I} = 8.0\times 10^{-4}\ \text{kg·m}^2.
$$

Assume the rod starts from rest at $t=0$ and the torque remains constant.

Answer the following:

1. Use $d\vec{F} = I\,d\vec{\ell}\times\vec{B}$ to derive the net force on the rod (direction and magnitude).
2. Compute the net torque about the pivot by integrating $d\tau_z = r\,dF$ from $x=0$ to $x=L$.
3. Compute the angular acceleration

   $$
   \alpha = \frac{\tau}{\mathcal{I}}.
   $$

4. How many **revolutions** does the rod complete in

   $$
   t = 1.5\ \text{s}
   $$

   assuming constant angular acceleration and starting from rest?
   Use

   $$
   \theta = \frac{1}{2}\alpha t^2,\qquad
   N = \frac{\theta}{2\pi}.
   $$

5. Briefly state one practical limitation of this idealized model (why real devices might not maintain constant $\alpha$).

---

<details>
<summary style="background-color:#006633; color:white; padding:8px; border-radius:4px; cursor:pointer;">
Problem solution
</summary>

<div style="background-color:#e8f5e9; padding:10px; border-radius:4px; margin-top:6px;">

**Relevant equations**

$$
d\vec{F}=I\,d\vec{\ell}\times\vec{B}
$$

$$
\tau_z = \int r\,dF
$$

$$
\alpha=\frac{\tau}{\mathcal{I}}
$$

$$
\theta=\frac{1}{2}\alpha t^2,\qquad N=\frac{\theta}{2\pi}
$$

---

**Given**

- $L=0.20\ \text{m}$
- $I=2.5\ \text{A}$
- $\vec{B}=0.60\,\hat{z}\ \text{T}$
- $\mathcal{I}=8.0\times 10^{-4}\ \text{kg·m}^2$
- $t=1.5\ \text{s}$, starts from rest

---

**1) Differential force and net force**

Along the rod:

$$
d\vec{\ell}=dx\,\hat{x}.
$$

Then

$$
d\vec{F}=I(dx\,\hat{x})\times(B\hat{z})
=IB\,dx\,(\hat{x}\times\hat{z}).
$$

Since $\hat{x}\times\hat{z}=-\hat{y}$,

$$
d\vec{F}=-IB\,dx\,\hat{y}.
$$

Net force:

$$
\vec{F}_{\text{net}}=\int_0^L d\vec{F}
=-IB\int_0^L dx\,\hat{y}
=-IBL\,\hat{y}.
$$

Magnitude:

$$
F_{\text{net}}=IBL=(2.5)(0.60)(0.20)=0.300\ \text{N}.
$$

So

$$
\boxed{\vec{F}_{\text{net}}=-0.300\,\hat{y}\ \text{N}}.
$$

---

**2) Net torque about the pivot**

A force element at position $x$ has lever arm $r=x$.  
Magnitude of the force element is $dF=IB\,dx$.

So

$$
d\tau_z = r\,dF = x(IB\,dx).
$$

Integrate:

$$
\tau_z=\int_0^L IBx\,dx
=IB\left[\frac{x^2}{2}\right]_0^L
=\frac{IBL^2}{2}.
$$

Compute:

$$
\tau_z=\frac{(2.5)(0.60)(0.20^2)}{2}
=\frac{(1.50)(0.040)}{2}
=0.0300\ \text{N·m}.
$$

Direction: $\vec{r}$ is $+\hat{x}$ and $d\vec{F}$ is $-\hat{y}$, so $\vec{\tau}$ is along $-\hat{z}$.

$$
\boxed{\vec{\tau}=-0.0300\,\hat{z}\ \text{N·m}}.
$$

---

**3) Angular acceleration**

Magnitude:

$$
\alpha=\frac{\tau}{\mathcal{I}}
=\frac{0.0300}{8.0\times 10^{-4}}
=37.5\ \text{rad/s}^2.
$$

$$
\boxed{\alpha=37.5\ \text{rad/s}^2}
$$

---

**4) Number of revolutions in $t=1.5\ \text{s}$**

Angular displacement from rest:

$$
\theta=\frac{1}{2}\alpha t^2
=\frac{1}{2}(37.5)(1.5^2).
$$

Since $1.5^2=2.25$,

$$
\theta=0.5(37.5)(2.25)=42.2\ \text{rad}.
$$

Revolutions:

$$
N=\frac{\theta}{2\pi}
=\frac{42.2}{2\pi}\approx 6.72.
$$

$$
\boxed{N\approx 6.7\ \text{revolutions in }1.5\ \text{s}}
$$

---

**5) One practical limitation**

Real devices usually do not maintain constant $\alpha$ because:

- friction and air drag add opposing torque,
- current can change as contacts heat up,
- the magnetic field may not be uniform,
- back-emf and circuit effects can reduce current in motor-like systems.

So the torque (and thus $\alpha$) typically decreases as speed increases.

</div>
</details>

---


In [ ]:
# DIY Cell

In [9]:
%reset -f

import sympy as sp
from IPython.display import display

# -----------------------------
# Given values
# -----------------------------
L = 0.20                 # m
I_current = 2.5          # A
B = 0.60                 # T  (B along +z)
I_rot = 8.0e-4           # kg·m^2  (moment of inertia)
t = 1.5                  # s

# -----------------------------
# Symbolic integration for force and torque
# -----------------------------
x = sp.symbols('x', real=True, nonnegative=True)

# Differential force magnitude on an element dx:
# dF = I * B * dx (since dℓ ⟂ B and sin(90°)=1)
dF = I_current * B

# Net force magnitude:
F_net = sp.integrate(dF, (x, 0, L))

# Net torque about pivot at x=0:
# dτ = x * dF * dx  -> τ = ∫_0^L x (I B) dx
tau = sp.integrate(x * dF, (x, 0, L))

# Angular acceleration:
alpha = tau / I_rot

# Angular displacement from rest with constant alpha:
theta = sp.Rational(1, 2) * alpha * t**2

# Number of revolutions:
N = theta / (2 * sp.pi)

# Numeric evaluation
F_net_num = float(F_net.evalf())
tau_num = float(tau.evalf())
alpha_num = float(alpha.evalf())
theta_num = float(theta.evalf())
N_num = float(N.evalf())

# -----------------------------
# Output
# -----------------------------
print("C5.4 — Magnetic actuator rod: torque, angular acceleration, and revolutions\n")

print("Given:")
print(f"  Rod length:            L = {L:.3e} m")
print(f"  Current:               I = {I_current:.3e} A")
print(f"  Magnetic field:        B = {B:.3e} T (along +z)")
print(f"  Moment of inertia:   ℑ = {I_rot:.3e} kg·m^2")
print(f"  Time interval:         t = {t:.3e} s\n")

print("Symbolic results:")
print("  Net force magnitude:   F = ∫(I B) dx from 0 to L")
display(F_net)

print("  Net torque magnitude:  τ = ∫ x (I B) dx from 0 to L")
display(tau)

print("\nNumeric results (4 significant figures):")
print(f"  Net force magnitude:   F = {F_net_num:.3e} N")
print(f"  Net torque magnitude:  τ = {tau_num:.3e} N·m")
print(f"  Angular acceleration:  α = {alpha_num:.3e} rad/s^2")
print(f"  Angular displacement:  θ = {theta_num:.3e} rad")
print(f"  Revolutions:           N = {N_num:.3e} rev\n")

print("Direction notes (from geometry):")
print("• With current along +x and B along +z, the force is along −y (since x̂ × ẑ = −ŷ).")
print("• With r along +x and F along −y, the torque is along −z (since x̂ × (−ŷ) = −ẑ).")


C5.4 — Magnetic actuator rod: torque, angular acceleration, and revolutions

Given:
  Rod length:            L = 2.000e-01 m
  Current:               I = 2.500e+00 A
  Magnetic field:        B = 6.000e-01 T (along +z)
  Moment of inertia:   ℑ = 8.000e-04 kg·m^2
  Time interval:         t = 1.500e+00 s

Symbolic results:
  Net force magnitude:   F = ∫(I B) dx from 0 to L


0.300000000000000

  Net torque magnitude:  τ = ∫ x (I B) dx from 0 to L


0.0300000000000000


Numeric results (4 significant figures):
  Net force magnitude:   F = 3.000e-01 N
  Net torque magnitude:  τ = 3.000e-02 N·m
  Angular acceleration:  α = 3.750e+01 rad/s^2
  Angular displacement:  θ = 4.219e+01 rad
  Revolutions:           N = 6.714e+00 rev

Direction notes (from geometry):
• With current along +x and B along +z, the force is along −y (since x̂ × ẑ = −ŷ).
• With r along +x and F along −y, the torque is along −z (since x̂ × (−ŷ) = −ẑ).


<hr style="height:5px;border-width:0;color:gray;background-color:gray">